### PR TITLE
Get parameters

### LINKED ISSUE #645: interface to `dist_spec` parameters
At the moment, accessing the parameters of a `dist_spec` is quite tedious, see e.g. https://github.com/epiforecasts/delayscompare/pull/3

It would be good to add a `get_parameters()` function to address this.

### PR DESCRIPTION
**Pull request by sbfnk**
## Description

This PR closes #645 by introducing `get_parameters()`, `get_pmf()` and `get_distribution()` functions that access elements of `<dist_spec>` objects.

## Initial submission checklist

- [X] My PR is based on a package issue and I have explicitly linked it.
- [X] I have tested my changes locally (using `devtools::test()` and `devtools::check()`).
- [X] I have added or updated unit tests where necessary.
- [X] I have updated the documentation if required and rebuilt docs if yes (using `devtools::document()`).
- [X] I have followed the established coding standards (and checked using `lintr::lint_package()`).
- [X] I have added a news item linked to this PR.

## After the initial Pull Request 

- [x] I have reviewed Checks for this PR and addressed any issues as far as I am able.

### CODE REVIEW

File: R/dist_spec.R [hunk 1]
```diff
@@ -1099,3 +1101,84 @@ convert_to_natural <- function(params, distribution) {
...
+  }
+  if (length(x) > 1) {
+    if (is.null(id)) {
+      stop("`id` must be specified when `x` is a composite distribution.")
+    }
+  } else {
+    id <- 1
+  }
+  return(id)
+}
```
Comment (jamesmbaazam) [inline R/dist_spec.R:1132]: This looks like it combines a "checking" with another functionality that makes an id? I suggest aligning this with how our other "checks" work, i.e., return an error if something is wrong, or invisibly return `x`. 

Maybe split this into a "check_id()" and "generate_id()". The latter will call the check function first and then return `id`.
 
Also, should this live in the `checks.R` script?

File: R/dist_spec.R [hunk 2]
```diff
@@ -1099,3 +1101,84 @@ convert_to_natural <- function(params, distribution) {
...
+##'
+##' @inheritParams check_get_dist_spec
+##' @description `r lifecycle::badge("experimental")`
+##' @return A list of parameters of the distribution.
+##' @export
+##' @examples
+##' dist <- Gamma(shape = 3, rate = 2)
+##' get_parameters(dist)
+get_parameters <- function(x, id = NULL) {
+  id <- check_get_dist_spec(x, id)
```
Comment (jamesmbaazam) [inline R/dist_spec.R:1144]: If my previous suggestion is taken, then this becomes two lines

```r
check_id(x, id)
id <- generate_id(x, id)
```

**Review by jamesmbaazam — APPROVED** [review]
Thanks, @sbfnk. The new accessors will make the distribution interface easier to interact with.

I've suggested a few changes that can be addressed or left for latter refactoring to move forward this release.

File: NEWS.md [hunk 3]
```diff
@@ -13,7 +13,8 @@
...
 * The function `init_cumulative_fit()` has been deprecated. By @jamesmbaazam in #541 and reviewed by @sbfnk.
-* The interface to generating delay distributions has been completely overhauled. Instead of calling `dist_spec()` users now specify distributions using functions that represent the available distributions, i.e. `LogNormal()`, `Gamma()` and `Fixed()`. Uncertainty is specified using calls of the same nature, to `Normal()`. More information on the underlying design can be found in `inst/dev/design_dist.md` By @sbfnk in #504 and reviewed by @seabbs.
+* The interface to generating delay distributions has been completely overhauled. Instead of calling `dist_spec()` users now specify distributions using functions that represent the available distributions, i.e. `LogNormal()`, `Gamma()` and `Fixed()`. Uncertainty is specified using calls of the same nature, to `Normal()`. More information on the underlying design can be found in `inst/dev/design_dist.md` By @sbfnk in #504 reviewed by @seabbs.
+* Accessor functions have been added to <dist_spec> options. By @sbfnk in .
```
Comment (jamesmbaazam) [inline NEWS.md:17]:
```suggestion
* Accessor functions have been added to <dist_spec> options. By @sbfnk in #646 and reviewed by @jamesmbaazam.
```

File: R/dist_spec.R (hunk 1 above)
Comment (sbfnk) [reply R/dist_spec.R]: Good point - I've renamed this to `get_dist_spec_id()` in [2b0565c](https://github.com/epiforecasts/EpiNow2/pull/646/commits/2b0565c6f9d15d59a6e41e1948874e2940350d5f). Could still ship out the first check but maybe it's fine in there?

**Review by jamesmbaazam — DISMISSED** [review]

File: R/dist_spec.R [hunk 4]
```diff
@@ -1099,3 +1101,86 @@ convert_to_natural <- function(params, distribution) {
...
+    stop("Can only get pmf of a <dist_spec>.")
+  }
+  id <- get_dist_spec_id(x, id)
+  if (x[[id]]$distribution != "nonparametric") {
+    stop("Cannot get pmf of a parametric distribution.")
+  }
+  return(x[[id]]$pmf)
+}
+
+##' Get the distribution of a [dist_spec()]
```
Comment (jamesmbaazam) [inline R/dist_spec.R:1171]:
```suggestion
##' Get the distribution of a `<dist_spec>`
```

File: R/dist_spec.R [hunk 5]
```diff
@@ -1099,3 +1101,86 @@ convert_to_natural <- function(params, distribution) {
...
+##' Perform checks for `<dist_spec>` `get_...` functions
+##'
+##' @param x A `<dist_spec>`.
+##' @param id Integer; the id of the distribution to get parameters of (if x is
+##' a composite distribution). If `x` is a single distribution this is ignored
+##' and can be left as `NULL`.
+##' @return The id to use.
+##' @keywords internal
+##' @author Sebastian Funk
+get_dist_spec_id <- function(x, id) {
```
Comment (jamesmbaazam) [inline R/dist_spec.R:1114]: The logic below suggests you mean to set `id = NULL` as the default right?

File: R/dist_spec.R [hunk 6]
```diff
@@ -1099,3 +1101,86 @@ convert_to_natural <- function(params, distribution) {
   }
   return(params)
 }
+
+##' Perform checks for `<dist_spec>` `get_...` functions
```
Comment (jamesmbaazam) [inline R/dist_spec.R:1105]:
```suggestion
##' Validate the "id" of a `<dist_spec>` `get_...()` function
```

**Review by jamesmbaazam — APPROVED** [review]

**Review by jamesmbaazam — APPROVED** [review]

**Pull request merged**